bit.py

```diff
@@ -1,70 +1,246 @@
+        return type(self)(complement_seq)
+
+    def reverse(self: Self) -> Self:
+        """
+        Returns the reversed sequence
+
+        Returns:
+            NucleicAcidSequence: a new sequence object with the reversed sequence
+        """
+        return type(self)(self.sequence[::-1])
+
+    def reverse_complement(self) -> Self:
+        """
+        Returns the reverse complement copy of sequence
+
+        Returns:
+            NucleicAcidSequence: a new sequence object containing the reversed complement sequence
+        """
+        self.reverse()
+        return self.complement()
+
+
+class DNASequence(NucleicAcidSequence):
+    """
+    Class for handling DNA sequences with the transcribe method
+    """
+
+    alphabet = "ATGCatgc"
+    complement_dict = {
+        "A": "T",
+        "T": "A",
+        "G": "C",
+        "C": "G",
+        "a": "t",
+        "t": "a",
+        "g": "c",
+        "c": "g",
+    }
+
+    def __init__(self: Self, sequence: str):
+        """
+        Initializes a DNASequence object
+
+        Args:
+            sequence (str): a string with the DNA sequence
+        """
+        super().__init__(sequence)
+
+    def transcribe(self: Self) -> Self:
+        """
+        Transcribes DNA into RNA (replaces "T" with "U")
+
+        Returns:
+            RNASequence: a new RNA sequence object
+        """
+        return RNASequence(self.sequence.replace("T", "U").replace("t", "u"))
+
+
+class RNASequence(NucleicAcidSequence):
+    """
+    Class for representing RNA sequences
+    """
+
+    alphabet = "AUGCaugc"
+    complement_dict = {
+        "A": "U",
+        "U": "A",
+        "G": "C",
+        "C": "G",
+        "a": "u",
+        "u": "a",
+        "g": "c",
+        "c": "g",
     }
 
-    res = function_map[operation](lst_nucl_seq)
-    if len(res) == 1:
-        return res[0]
-    return res
+    def __init__(self: Self, sequence: str):
+        """
+        Initializes an RNASequence object.
+
+        Args:
+            sequence (str): a string with the RNA sequence
+        """
+        super().__init__(sequence)
+
+
+class AminoAcidSequence(BioSequenceFunctions):
+    """
+    Class for representing amino acid sequences with methods for counting amino acids in sequence
+    """
+
+    alphabet = "GAVLITSMCPFYWHKRDENQgavlitsmcpfywhkrdenq"
+
+    def __init__(self: Self, sequence: str):
+        """
+        Initializes an AminoAcidSequence object
+
+        Args:
+            sequence (str): a string representing the amino acid sequence
+        """
+        super().__init__(sequence)
+
+    def count_aa(self: Self) -> dict:
+        """
+        Counts the occurrences of each amino acid in the sequence
+
+        Returns:
+            dict: a dictionary with amino acids as keys and their counts as values
+        """
+        counter = dict()
+        for aa in self.sequence:
+            if aa not in counter.keys():
```

Comment:
```suggestion
            if aa not in counter:
```

Итерация по словарю это и так итерация по ключам по умолчанию.